enhancements/implicit-catalog-versioning.md

```diff
@@ -150,3 +150,182 @@ There was some significant discussion around bumping the CSV API version to `v1a
+                    "csv": "etcdoperator.v0.9.0",
+                    "bundlePath": "quay.io/etcd/etcd-operator-bundle@sha256:defghi",
+                    "replaces": [
+                        {
+                            "version": "0.6.1",
+                            "csv": "etcdoperator.v0.6.1"
+                        }
+                    ],
+                    "replacements": [
+                        {
+                            "version": "0.9.2",
+                            "csv": "etcdoperator.v0.9.2"
+                        }
+                    ]
+                },
+                {
+                    "version": "0.9.2",
+                    "csv": "etcdoperator.v0.9.2",
+                    "bundlePath": "quay.io/etcd/etcd-operator-bundle@sha256:ghijkl",
+                    "replaces": [
+                    {
+                        "version": "0.6.1",
+                        "csv": "etcdoperator.v0.6.1"
+                    },
+                    {
+                        "version": "0.9.0",
+                        "csv": "etcdoperator.v0.9.0"
+                    }
+                    ],
+                    "replacements": null
+                }
+            ]
+        },
+        {
+            "name": "stable",
+            "bundles": [
+                {
+                    "version": "0.9.0",
+                    "csv": "etcdoperator.v0.9.0",
+                    "bundlePath": "quay.io/etcd/etcd-operator-bundle@sha256:defghi",
+                    "replaces": null,
+                    "replacements": [
+                        {
+                            "version": "0.9.2",
+                            "csv": "etcdoperator.v0.9.2"
+                        }
+                    ]
+                },
+                {
+                    "version": "0.9.2",
+                    "csv": "etcdoperator.v0.9.2",
+                    "bundlePath": "quay.io/etcd/etcd-operator-bundle@sha256:ghijkl",
+                    "replaces": [
+                        {
+                            "version": "0.9.0",
+                            "csv": "etcdoperator.v0.9.0"
+                        }
+                    ],
+                    "replacements": null
+                }
+            ]
+        }
+    ]
+}
+```
+
+Will return a list of bundles along with the bundle image they were built from and lists of (zero or more) bundles that they replace and are replaced by. One thing to note about this result is that, in it's current state `version` could still be null at this time. In the future, if CSV-less bundles become a common standard that requirement may be updated where CSV becomes an optional field and version becomes required.
+
+Additionally, now that semantic version is used to insert bundles anywhere in the graph, we can also delete individual bundles in any part of the graph as long as we are in `--semver` or `--semver-skippatch` mode. We will create a new command `opm index rmb`:
```

Comment:
why not `opm index remove` to go with `opm index add`?

Reply:
`opm index rm` already exists and is used to delete whole packages. This is sort of similar to the `docker rm` vs `docker rmi` syntax.

Reply:
I don't think the docker cli experience is very intuitive.

maybe we can use the same command and switch between package and bundle based on an additional qualifier:

```
# Remove bundle by name
opm index rm bundle:foo
# Remove bundle by checking if it's a valid image reference
opm index rm https://quay.io/foo@sha256:....
# Remove package
opm index rm package:bar
```